components/post_view/message_attachments/action_button/action_button.test.jsx

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+import {changeOpacity} from 'mattermost-redux/utils/theme_utils';
+
+import {Constants, STATUS_COLORS} from 'utils/constants';
+import ActionButton from 'components/post_view/message_attachments/action_button/action_button';
+
+describe('components/post_view/message_attachments/action_button.jsx', () => {
+    const baseProps = {
+        action: {id: 'action_id_1', name: 'action_name_1', cookie: 'cookie-contents'},
+        handleAction: jest.fn(),
+        theme: Constants.THEMES.default,
+    };
+
+    test('should match snapshot', () => {
+        const wrapper = shallow(<ActionButton {...baseProps}/>);
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should call handleAction on click', () => {
+        const wrapper = shallow(<ActionButton {...baseProps}/>);
+
+        wrapper.find('button').simulate('click');
+
+        expect(baseProps.handleAction).toHaveBeenCalledTimes(1);
+    });
+
+    test('should have correct styles when provided color from theme', () => {
```

Comment:
Can you add a test that changes the `theme` property and asserts the requisite behaviour? (See comment above re: defaulting to the default theme always.)

Reply:
Done :+1: